src/server.ts

```diff
@@ -46,6 +47,32 @@ const startServer = async () => {
     return 'This is the response\n'
   })
 
+  // Unique name/email/organisation check
+  server.get('/check-unique', async (request: any, reply) => {
+    const type = request.query.type
+    const value = request.query.value
+    if (value === '' || value === undefined) return false
+    let table, field
+    switch (type) {
+      case 'username':
+        table = 'user'
+        field = 'username'
+        break
+      case 'email':
+        table = 'user'
+        field = 'email'
+        break
+      case 'organisation':
+        table = 'organisation'
+        field = 'name'
+        break
+      default:
```

Comment:
I think it would be good to have some alert if some type passed is not treated, so basically send back the result with result + message. It can still return false, but the message would say: `not a recognised type`.

Reply:
How do you suggest structuring this? Currently, the endpoint always returns a boolean, so are you suggesting returning another type (like an object) as the result for those situations?

Reply:
Yes. I think it would be nicer to have the result as an object and the queryResult is returned in the body. It's also preferable to return the status in the api call as what can be checked first and then the result in the body. For example for success = `{ status: 200, resultBody = { unique: "false" }`. That's basically my opinion from what I have seen before, not 100% certain. what do you think @andreievg?

Reply:
I don't think we need the status as that's included in the HTTP response headers anyway.

I agree that returning a pure boolean is probably not best practice in a general sense. Only problem is that in this case writing a dynamic expression to parse the return object will make for a more complex query. Or maybe it won't be too bad, will have to try and see. How about a response like this:
```
{ 
  unique: true/false
  message: "Not a recognised type" (or <empty string> if type is okay)
}
```

I'll try that and then re-write the email validation query and see how much more complex it is.

Reply:
Okay, I've changed it to the above and it didn't actually make the queries much more complex (just one extra `{value: "unique"}` node in the API lookup :) I've updated the API tests in #68 to reflect this change too.

Let me know if that works for you.

Reply:
> I don't think we need the status as that's included in the HTTP response headers anyway.

That's what I was trying to say. It looks like an object though. It was supposed to be a wrapper of what is returned by the API.

> :) I've updated the API tests in #68 to reflect this change too.

Thanks! Will have a look.